tests/core/test_websocket_peer.h

```diff
@@ -0,0 +1,49 @@
+/**************************************************************************/
+/* test_websocket_peer.h                                                  */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+
+#include "modules/websocket/websocket_peer.h"
+#include "tests/test_macros.h"
+
+namespace TestWebSocketPeer {
+
+TEST_CASE("[WebSocketPeer] after creation state is closed") {
+	WebSocketPeer *peer = WebSocketPeer::create();
+	CHECK_MESSAGE(peer != nullptr, "WebSocketPeer::create() returned nullptr");
+
+	if (peer) {
+		CHECK(peer->get_ready_state() == WebSocketPeer::STATE_CLOSED);
+		memdelete(peer);
+	}
+
```

Comment:
```suggestion
	Ref<WebSocketPeer> peer = WebSocketPeer::create();
	REQUIRE_MESSAGE(peer.is_valid(), "WebSocketPeer::create() returned nullptr.");

	CHECK(peer->get_ready_state() == WebSocketPeer::STATE_CLOSED);
```
`WebSocketPeer` is `RefCounted` so should use `Ref` semantics, also the check for `peer` is unnecessary since we already checked above, if using `REQUIRE`

Reply:
@lilcfar Can you look into applying the suggestion?

Once this is done, please squash the commits together so we can merge this. See [PR workflow](https://contributing.godotengine.org/en/latest/organization/pull_requests/creating_pull_requests.html#the-interactive-rebase) for instructions :slightly_smiling_face: